modules/decrypt-node/src/parse_header_stream.ts

```diff
@@ -36,14 +44,34 @@ export class ParseHeaderStream extends PortableTransformWithType {
     this._headerState = {
       buffer: Buffer.alloc(0),
       headerParsed: false,
+      maxHeaderSize: maxHeaderSize,
     }
   }
 
   _transform(chunk: any, encoding: string, callback: Function) {
-    const { buffer } = this._headerState
+    const { buffer, maxHeaderSize } = this._headerState
     const headerBuffer = Buffer.concat([buffer, chunk])
-    const headerInfo = deserialize.deserializeMessageHeader(headerBuffer)
+
+    /* maxHeaderSize is the maximum
+     * size we MUST attempt
+     * to deserialize.
+     */
+    const headerInfo = deserialize.deserializeMessageHeader(
+      maxHeaderSize && headerBuffer.byteLength > maxHeaderSize
+        ? headerBuffer.slice(0, maxHeaderSize)
+        : headerBuffer
```

Comment:
Suggestion: rather than passing in a slice sometimes and sometimes the original, what if we always pass in a slice and just change where the end of the slice is?